make_BAF_igv.py

```diff
@@ -4,57 +4,79 @@
             chrom = record.CHROM
             end = record.POS
             start = end - 1
             locus = "{}:{}-{}".format(chrom, start, end)
-            try:
+
+            if 'DP' in record.genotype(sampleid).data._asdict():
                 dp = record.genotype(sampleid)['DP']
-            except:  # Skip ref/ref cals without AD field (this does exist)
-                print("Locus {} has no DP field, skipping variant position as BAF could not be determined".format(locus))
+            else: # Skip variant position as BAF could not be determined because DP is missing
                 continue
+
             variant_call = record.genotype(sampleid).is_variant  # Returns True if a variant-call, and thus not a reference-call
             if dp >= args.mindepth:
                 if variant_call == True: # Calculate ad and baf for variant-calls
                     ad = record.genotype(sampleid)['AD']
                     baf = (ad[1] / dp) * 100
                 elif variant_call == False: # Calculate ad and baf for reference-calls
-                    try:
+
+                    if 'AD' in record.genotype(sampleid).data._asdict():
                         ad = [record.genotype(sampleid)['AD']]
                         baf = ((dp - ad[0]) / (dp)) * 100
-                    except:  # Skip ref/ref cals without AD field (this does exist)
-                        print("Locus {} has no AD field, skipping variant position as BAF could not be determined".format(locus))
+                    else: # Skip variant position as BAF could not be determined because AP is missing
                         continue
+
                 else: # In case variant status is not known (no call or ./.)
                     continue
             else:  # If threshold if not met, skip position
                 continue
 
             if len(ad) <= 2:  # skip multiallelic sites
-                output_file.write("{chrom}\t{start}\t{end}\t{locus}\t{baf:.2f}\n".format(
+                variant_line = "{chrom}\t{start}\t{end}\t{locus}\t{baf:.2f}".format(
                     chrom=chrom,
                     start=start,
                     end=end,
                     locus=locus,
                     baf=baf
-                    ))
+                    )
+
+                if args.outputfile:
+                    output_file.write("{}\n".format(variant_line))
+                else:
+                    print(variant_line)
 
-    output_file.close()
+    if args.outputfile:
+        output_file.close()
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument('inputfile', help='input VCF file (genotyped VCF)')
-    parser.add_argument('outputfile', help='output filename')
+    parser.add_argument('inputfile', help='input VCF file (genotyped VCF). VCF must be uncompressed')
+    parser.add_argument('-o', '--outputfile', help='output filename')
```

Comment:
you told me that all these arguments are positional and therefore required. but your changes in the code makes "outputfile" not required anymore?

Reply:
I did mention it somewhere:  default is now stdout and outputfile is now optional.

Added clarification to help.

Reply:
you showed me your current help function, and it's indeed now in "optional arguments". I dont fully understand it, but it works now :+1: